### PR TITLE
Some mixed missed Equals nullable annotations 

### PR DESCRIPTION
**Pull request by hrrrrustic**
Decided to push it without splitting by areas because there are not so much changes

### CODE REVIEW

**Comment by hrrrrustic** [comment]
```C#
public override bool Equals([System.Diagnostics.CodeAnalysis.NotNullWhenAttribute(true)] object? obj) { throw null; }
```
> The type or namespace name 'NotNullWhenAttributeAttribute' does not exist in the namespace 'System.Diagnostics.CodeAnalysis' (are you missing an assembly reference?)

Why "Attribute" was repeated?

**Comment by jeffhandley** [comment]
> ```cs
> public override bool Equals([System.Diagnostics.CodeAnalysis.NotNullWhenAttribute(true)] object? obj) { throw null; }
> ```
> 
> > The type or namespace name 'NotNullWhenAttributeAttribute' does not exist in the namespace 'System.Diagnostics.CodeAnalysis' (are you missing an assembly reference?)
> 
> Why "Attribute" was repeated twice?

Weird -- I don't understand what's happening here either. @krwq can you review this and chime in if you've seen this before?

**Comment by stephentoub** [comment]
> Weird -- I don't understand what's happening here either.

The compiler is searching for the type,  both as written and with an Attribute suffix added on (even though it already has one). It means the attribute isn't imported,  which generally means either a) a using is missing or b) it's coming from some build targeting a framework that lacks the attribute and where we're not yet injecting an internal version.

**Comment by jeffhandley** [comment]
Thanks, @stephentoub!

@hrrrrustic -- let us know if you need any help narrowing the build errors down from here.

**Comment by jeffhandley** [comment]
Once this is merged, we'll need to edit the description on dotnet/docs#21202 to include the affected APIs from this PR.

**Comment by hrrrrustic** [comment]
> a) a using is missing

What kind of using you're talking about? The error was in the ref file, I've never saw usings in it 😄 
I tried to add
```
</ItemGroup>
    <ItemGroup>
    <ProjectReference Include="..\..\System.Runtime\ref\System.Runtime.csproj" />
</ItemGroup>
```
to `Microsoft.Extensions.Logging.Abstractions.csproj`, but got 
`error NU1201: Project System.Runtime is not compatible with netstandard2.0 (.NETStandard,Version=v2.0). Project System.Runtime supports: net6.0 (.NETCoreApp,Version=v6.0)`
and
`error NU1201: Project System.Runtime is not compatible with net461 (.NETFramework,Version=v4.6.1). Project System.Runtime supports: net6.0 (.NETCoreApp,Version=v6.0)`

So I assume it's not possible to use this attribute here yet (correct me if I'm wrong) and simply revert changes

**Comment by buyaa-n** [comment]
@hrrrrustic `NullableAttributes.cs` is not included in `.Net Framework`, probably in `netstandard` too, seems you need to add the file like this 
https://github.com/dotnet/runtime/blob/aad916c7111323d4b5744eb2444252d149c57a8a/src/tasks/WasmAppBuilder/WasmAppBuilder.csproj#L13

**Comment by hrrrrustic** [comment]
Rebased on top to be above #53199

File: src/libraries/Microsoft.Extensions.Logging.Abstractions/ref/Microsoft.Extensions.Logging.Abstractions.csproj [hunk 1]
```diff
@@ -5,5 +5,6 @@
   </PropertyGroup>
   <ItemGroup>
     <Compile Include="Microsoft.Extensions.Logging.Abstractions.cs" />
+    <Compile Include="$(RepoRoot)src\libraries\System.Private.CoreLib/src/System/Diagnostics/CodeAnalysis/NullableAttributes.cs" />
```
Comment (buyaa-n) [inline src/libraries/Microsoft.Extensions.Logging.Abstractions/ref/Microsoft.Extensions.Logging.Abstractions.csproj:8]: Ah sorry @hrrrrustic seems `<Nullable>enable</Nullable>` above was enough, seems after that added the `NullableAttributes` getting added automatically, please remove this line
```suggestion
```

**Review by buyaa-n — APPROVED** [review]
LGTM, thanks!

**Comment by buyaa-n** [comment]
The Wasm test failures seem [fixed ](https://github.com/dotnet/runtime/issues/53405), but rerunning that CI leg is not fixing the failure, so closing the PR and reopening it back

```log
Wasm.Build.Tests.NativeBuildTests.Relinking_ErrorWhenMissingEMSDK [FAIL]
      System.TypeInitializationException : The type initializer for 'Wasm.Build.Tests.BuildTestBase' threw an exception.
      ---- System.TypeLoadException : Could not load type 'System.IO.DirectoryInfo' from assembly 'System.Runtime, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'.
      Stack Trace:
        /_/src/tests/BuildWasmApps/Wasm.Build.Tests/BuildTestBase.cs(123,0): at Wasm.Build.Tests.BuildTestBase.ConfigWithAOTData(Boolean aot)
        /_/src/tests/BuildWasmApps/Wasm.Build.Tests/InvariantGlobalizationTests.cs(21,0): at Wasm.Build.Tests.InvariantGlobalizationTests.InvariantGlobalizationTestData(Boolean aot, RunHost host)
```

**Pull request merged**